tools/codegen/dest/register_dispatch_key.py

```diff
@@ -73,17 +73,13 @@ def gen_resize_out_helper(backend_index: BackendIndex) -> List[str]:
       "Expected out tensor to have dtype ", options.dtype(), ", but got ", out.dtype(), " instead");
   TORCH_CHECK(options.device() == out.device(),
       "Expected out tensor to have device ", options.device(), ", but got ", out.device(), " instead");
-  const bool resized = at::native::resize_output(out, sizes);
-  // Only restride if a resize occurred; otherwise we ignore the (advisory)
-  // strides from the meta function and directly use the output tensor's
-  // preexisting strides
-  if (resized) {
-    if (!strides.empty()) {
-      TORCH_INTERNAL_ASSERT(!options.memory_format_opt().has_value());
-      at::native::as_strided_(out, sizes, strides);
-    } else if (options.memory_format_opt().has_value()) {
-      out.unsafeGetTensorImpl()->empty_tensor_restride(*options.memory_format_opt());
-    }
+  c10::optional<IntArrayRef> expanded_strides = strides.empty() ? c10::nullopt
+      : c10::optional<IntArrayRef>(strides);
+  const bool resized = at::native::resize_output(out, sizes, expanded_strides);
```

Comment:
Is this for @Lezcano 's proposal? If we call something like:
```
a = contiguous_tensor()  # row major strides
torch.linalg_op(..., out=a) # suppose a has the right size, but the linalg_op needs to operate on column major strides
```

Then I don't think we want to actually modify `a`'s strides. out= ops shouldn't modify the metadata on the out= tensor; instead, we need to construct an intermediate tensor with the right strides and pass that into the IMPL func, and then copy back the result to the user's out= tensor.

That's my understanding of @Lezcano's idea. I think it should be done in a separate PR, since it's pretty independent from the `at::sort` issue addressed in this stack (and won't require changes to `resize_()`, but will require changes to the structured kernel codegen). Let me know if you agree!

Reply:
That was exactly my idea, yep. I also agree that the strides of the out tensor should not be changed.

Reply:
> Is this for @Lezcano 's proposal?

No, it was for my previous proposal.

> out= ops shouldn't modify the sizes/strides on the out= tensor; 

Ideally, right!? Currently, it does modify the sizes, even though we do throw a warning.

> I think it should be done in a separate PR

That makes sense. So, I'm going to close both PRs in this stack and open a new one for `at::sort`. I'm also opening a new issue just for tracking @Lezcano idea.